apm-agent-core/src/test/java/specs/ApiKeysStepsDefinitions.java

```diff
@@ -0,0 +1,160 @@
+package specs;
+
+import co.elastic.apm.agent.configuration.converter.TimeDuration;
+import co.elastic.apm.agent.report.ApmServerClient;
+import co.elastic.apm.agent.report.HttpUtils;
+import co.elastic.apm.agent.report.ReporterConfiguration;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+import com.github.tomakehurst.wiremock.extension.responsetemplating.ResponseTemplateTransformer;
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.stagemonitor.configuration.ConfigurationOption;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collections;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Fail.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ApiKeysStepsDefinitions {
```

Comment:
The step definitions look very nice and clean!

Reply:
Yes, I like it too, however I feel that the shared spec we could be further improved to remove similar and not explicit statements, I'll open another PR for this.

Reply:
The PR has already been merged on shared spec improvement. https://github.com/elastic/apm/pull/226